Source/NSOrderedSet.m

```diff
@@ -0,0 +1,1903 @@
+{
+  return AUTORELEASE([[self allocWithZone: NSDefaultMallocZone()]
+		       initWithObjects: objects count: count]);
+}
+
++ (instancetype) orderedSetWithOrderedSet:(NSOrderedSet *)aSet
+{
+  return AUTORELEASE([[self allocWithZone: NSDefaultMallocZone()]
+		       initWithOrderedSet: aSet]);
+}
+
+
++ (instancetype) orderedSetWithSet:(NSSet *)aSet
+{
+  return AUTORELEASE([[self allocWithZone: NSDefaultMallocZone()]
+		       initWithSet: aSet]);
+}
+
++ (instancetype) orderedSetWithSet:(NSSet *)aSet
+                         copyItems:(BOOL)flag
+{
+  return AUTORELEASE([[self allocWithZone: NSDefaultMallocZone()]
+			initWithSet: aSet
+			  copyItems: flag]);
+}
+
+// instance methods
+- (instancetype) initWithArray:(NSArray *)other
+{
+  unsigned	count = [other count];
+
+  if (count == 0)
+    {
+      return [self init];
+    }
+  else
+    {
+      GS_BEGINIDBUF(objs, count);
+
+      if ([other isProxy])
+	{
+	  unsigned	i;
+
+	  for (i = 0; i < count; i++)
+	    {
+	      objs[i] = [other objectAtIndex: i];
+	    }
+	}
+      else
+	{
+          [other getObjects: objs];
+	}
+      self = [self initWithObjects: objs count: count];
+      GS_ENDIDBUF();
+      return self;
+    }
+
+  return nil;
+}
+
+- (instancetype) initWithArray:(NSArray *)other copyItems:(BOOL)flag
+{
+  unsigned	count = [other count];
+  unsigned      j = count;
+  
+  if (count == 0)
+    {
+      return [self init];
+    }
+  
+  GS_BEGINIDBUF(objs, count);
+  {
+    unsigned	i;
+    
+    for (i = 0; i < count; i++)
+      {
+	if (flag == NO)
+	  {
+	    objs[i] = [other objectAtIndex: i];
+	  }
+	else // copy the items.
+	  {
+	    objs[i] = [[other objectAtIndex: i] copy];
+	  }
+      }
+  }
+  
+  self = [self initWithObjects: objs count: count];
+  
+  if(flag == YES)
+    {
+      while(j--)
+	{
+	  [objs[j] release];
+	}
+    }
+  GS_ENDIDBUF();
+  return self;
+}
+
+- (instancetype) initWithArray:(NSArray *)other
+                         range:(NSRange)range
+                     copyItems:(BOOL)flag
+{
+  unsigned	count = [other count];
+  unsigned      i = 0, j = 0;
+    
+  if (count == 0)
+    {
+      return [self init];
+    }
+
+  GS_BEGINIDBUF(objs, range.length);
+  {
+    unsigned      loc = range.location;
+    unsigned      len = range.length;
+    
+    for (i = 0; i < count; i++)
+      {
+	if(i >= loc && j < len)
+	  {
+	    if(flag == YES)
+	      {		  
+		objs[i] = [[other objectAtIndex: i] copy];
+	      }
+	    else
+	      {
+		objs[i] = [other objectAtIndex: i];
+	      }
+	    j++;
+	  }
+	
+	if(j >= len)
+	  {
+	    break;
+	  }
+      }
+  }
+  self = [self initWithObjects: objs count: count];
+
+  if(flag == YES)
+    {
+      while(j--)
+	{
+	  [objs[j] release];
+	}
+    }
+  GS_ENDIDBUF();
+
+  return self;
+}
+
+- (instancetype) initWithObject: (id)obj
+{
+  id objs[] = {obj};
+  
+  self = [self initWithObjects: objs count: 1];
+  if(self == nil)
+    {
+      NSLog(@"Problem initializing with one element");
```

Comment:
Do we really want this log message? In all other cases we just hand on the result.